src/ffs/public/util.ts

```diff
@@ -40,15 +40,16 @@ export const addRecurse = async (tree: Tree, path: NonEmptyPath, child: Tree | F
   return tree.updateDirectChild(toAdd, name)
 }
 
-export const getRecurse = async (tree: Tree, path: string[]): Promise<Tree | File> => {
-  const nextTree = await tree.getDirectChild(path[0])
-  if(path.length <= 1 && nextTree !== null){
+export const getRecurse = async (tree: Tree, path: NonEmptyPath): Promise<Tree | File | null> => {
+  const head = path[0]
+  const nextPath = pathUtil.nextNonEmpty(path)
+  const nextTree = await tree.getDirectChild(head)
+  if(nextPath === null){
     return nextTree
-  }
-  if(nextTree === null || isFile(nextTree)){
-    throw new Error("Path does not exist")
-  }
-  return getRecurse(nextTree, path.slice(1))
+  } else if (nextTree === null || isFile(nextTree)){
+    return null
```

Comment:
Shouldn't this be `return nextTree`? 🤔

Reply:
Or does it only return a file when `nextPath === null`?

Reply:
yup only when `nextPath === `null`
since if there is still some path to go down, and the "next tree" is either a file or doesn't exist, it's impossible to do so